packages/store-sync/src/wiresaw.ts

```diff
@@ -16,70 +18,98 @@ type WatchLogsResult = {
 };
 
 export function watchLogs({ url, address, fromBlock }: WatchLogsInput): WatchLogsResult {
-  // Buffer the live logs received until the gap from `startBlock` to `currentBlock` is closed
-  let caughtUp = false;
-  const logBuffer: StoreEventsLog[] = [];
-
   const topics = [
     storeEventsAbi.flatMap((event) => encodeEventTopics({ abi: [event], eventName: event.name })),
   ] as LogTopic[]; // https://github.com/wevm/viem/blob/63a5ac86eb9a2962f7323b4cc76ef54f9f5ef7ed/src/actions/public/getLogs.ts#L171
 
+  let resumeBlock = fromBlock;
+  let keepAliveInterval: ReturnType<typeof setTimeout> | undefined = undefined;
   const logs$ = new Observable<StorageAdapterBlock>((subscriber) => {
     let client: SocketRpcClient<WebSocket>;
-    getWebSocketRpcClient(url, { keepAlive: true, reconnect: true }).then(async (_client) => {
-      client = _client;
-      client.socket.addEventListener("error", (error) =>
-        subscriber.error({ code: -32603, message: "WebSocket error", data: error }),
-      );
-
-      // Start watching pending logs
-      const subscriptionId: Hex = (
-        await client.requestAsync({
-          body: {
-            method: "wiresaw_watchLogs",
-            params: [{ address, topics }],
-          },
-        })
-      ).result;
-
-      // Listen for wiresaw_watchLogs subscription
-      // Need to use low level methods since viem's socekt client only handles `eth_subscription` messages.
-      // (https://github.com/wevm/viem/blob/f81d497f2afc11b9b81a79057d1f797694b69793/src/utils/rpc/socket.ts#L178)
-      client.socket.addEventListener("message", (message) => {
-        const response = JSON.parse(message.data);
-        if ("error" in response) {
-          // Return JSON-RPC errors to the subscriber
-          subscriber.error(response.error);
-          return;
-        }
 
-        // Parse the logs from wiresaw_watchLogs
-        if ("params" in response && response.params.subscription === subscriptionId) {
-          const logs: RpcLog[] = response.params.result;
-          const formattedLogs = logs.map((log) => formatLog(log));
-          const parsedLogs = parseEventLogs({ abi: storeEventsAbi, logs: formattedLogs });
-          if (caughtUp) {
-            const blockNumber = parsedLogs[0].blockNumber;
-            subscriber.next({ blockNumber, logs: parsedLogs });
-          } else {
-            logBuffer.push(...parsedLogs);
+    function setupClient(): void {
+      // Buffer the live logs received until the gap from `startBlock` to `currentBlock` is closed
+      let caughtUp = false;
+      const logBuffer: StoreEventsLog[] = [];
+
+      getWebSocketRpcClient(url).then(async (_client) => {
```

Comment:
looks like this is a fairly lightweight wrapper around `getSocketRpcClient` that implements the ping and such: https://github.com/wevm/viem/blob/main/src/utils/rpc/webSocket.ts

wonder if we should use this instead? and replace their ping implementation?

Reply:
We would also have to reimplement the remaining stuff in `getWebSocketRpcClient` no? We could also just pass `keepAlive: false` sice keep alive is handled outside, but still keep the reconnect and other logic in `getWebSocketRpcClient`

Reply:
was thinking we could let the keep alive stay as is and not reimplement that and instead do something different in the `ping` method

but I am seeing now that lower level stuff doesn't do the request/response lifecycle, just sends stuff to the socket and something else handles data coming back, so maybe not a great candidate for it